File: Source/Lib/RCCService/Types/ThumbnailSettings.ts
import { ThumbnailCommandType } from './ThumbnailCommandType';

export class ThumbnailSettings {
	public Type: ThumbnailCommandType;

	public Arguments: any[];

	public constructor(type: ThumbnailCommandType, args: any[]) {
		this.Type = <ThumbnailCommandType>(<unknown>ThumbnailSettings.ThumbnailCommandTypeToString(type));
		this.Arguments = args;
	}

	private static ThumbnailCommandTypeToString(type: ThumbnailCommandType) {
		return ThumbnailCommandType[type];
	}
}

File: lib/grid/commands/Settings/ThumbnailSettings.cs
namespace MFDLabs.Grid.Commands;

/// <summary>
/// Settings for <see cref="ThumbnailCommand"/>
/// </summary>
public class ThumbnailSettings
{
    /// <summary>
    /// The type of thumbnail command.
    /// </summary>
    public ThumbnailCommandType Type { get; }

    /// <summary>
    /// The arguments to pass to the thumbnail command.
    /// </summary>
    public object[] Arguments { get; }

    /// <summary>
    /// Construct a new instance of <see cref="ThumbnailSettings"/>
    /// </summary>
    /// <param name="type">The type of thumbnail command.</param>
    /// <param name="arguments">The arguments to pass to the thumbnail command.</param>
    public ThumbnailSettings(ThumbnailCommandType type, params object[] arguments)
    {
        Type = type;
        Arguments = arguments;
    }
}
